auth/basicauthenticator/basicauth.go

```diff
@@ -0,0 +1,35 @@
+package basicauthenticator
```

Comment:
can we name this auth/internal/authenticator.go? While it does use basic auth, other providers can use it too, for obtaining credentials.

Reply:
`/internal/` in the import path would prevent other packages to import this. I am not quite sure on what you want to achieve here. Could you please explain further? Thanks